core/go/internal/publictxmgr/in_flight_transaction_state_version.go

```diff
@@ -0,0 +1,340 @@
+/*
+ * Copyright © 2025 Kaleido, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package publictxmgr
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/kaleido-io/paladin/core/internal/msgs"
+	"github.com/kaleido-io/paladin/core/pkg/ethclient"
+	"github.com/kaleido-io/paladin/toolkit/pkg/i18n"
+	"github.com/kaleido-io/paladin/toolkit/pkg/log"
+	"github.com/kaleido-io/paladin/toolkit/pkg/pldapi"
+	"github.com/kaleido-io/paladin/toolkit/pkg/tktypes"
+)
+
+type inFlightTransactionStateVersion struct {
+	// This id is never persisted in the database it only exists in memory so that we can pass around a reference
+	// to the version id in some functions rather than passing the reference to the version struct itself. This
+	// is because having any types imported from this package into mocks files results in circular dependencies
+	id int
+
+	current             bool
+	testOnlyNoEventMode bool
+
+	PublicTxManagerMetricsManager
```

Comment:
Just an observation that the nesting (rather than reference) of many parent objects in a structure, is quite a unique style to this part of Paladin.

It's very convenient to be able to reference fields, but the counter I've seen is that when you're referring to a field it's sometimes hard to know the context/ownership of the data to which you're referring.

No specific pressure to change that, but I wanted to give the perspective for you to take or leave.